frontend/src/components/ui/sidebar.tsx

```diff
@@ -0,0 +1,771 @@
+import { Slot } from "@radix-ui/react-slot";
+import { VariantProps, cva } from "class-variance-authority";
+import { Columns2Icon } from "lucide-react";
+import * as React from "react";
+import { Button } from "src/components/ui/button";
+import { Input } from "src/components/ui/input";
+import { Separator } from "src/components/ui/separator";
+import {
+  Sheet,
+  SheetContent,
+  SheetDescription,
+  SheetHeader,
+  SheetTitle,
+} from "src/components/ui/sheet";
+import { Skeleton } from "src/components/ui/skeleton";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "src/components/ui/tooltip";
+import { useIsMobile } from "src/hooks/use-mobile";
+import { cn } from "src/lib/utils";
+
+const SIDEBAR_COOKIE_NAME = "sidebar_state";
```

Comment:
this doesn't seem to work and not sure we want to save a cookie?

Reply:
changed it to localstorage and made it work :ok: